src/bridge/mod.rs

```diff
@@ -188,16 +187,13 @@ async fn start_neovim_runtime(
         close_watcher_running.store(false, Ordering::Relaxed);
     });
 
-    if let Ok(output) = nvim.command_output("version").await {
-        let re = Regex::new(r"NVIM v0.[4-9]\d*.\d+").unwrap();
-        if !re.is_match(&output) {
+    match nvim.command_output("echo has('nvim-0.4')").await.as_deref() {
+        Ok("1") => {} // This is just a guard
+        _ => {
             error!("Neovide requires nvim version 0.4 or higher. Download the latest version here https://github.com/neovim/neovim/wiki/Installing-Neovim");
             std::process::exit(0);
         }
```

Comment:
```rust
if !nvim.command_output("echo has('nvim-0.4')").await.map_or(false, |i| i ==1) {
    error!(...)
    ...
}
```

Kinda feels like there should be a `nvim.has` function.

Reply:
I remember messing with this area of the code to make it a bit more robust for future versions of nvim. I'll look at it with fresher eyes tomorrow. The code was failing before my change because some people were installing 0.5.0 when we only technically supported 0.4.0 I think.

Reply:
What versions of nvim are intended to be supported? This check (the regex one) accepts versions 0.4 to 0.9, but not version 0.10 (or 1.0). It also (probably accidentally) accepts versions 0.40 to 0.99.

> I remember messing with this area of the code to make it a bit more robust for future versions of nvim. I'll look at it with fresher eyes tomorrow. The code was failing before my change because some people were installing 0.5.0 when we only technically supported 0.4.0 I think.

[Here](https://github.com/neovide/neovide/pull/631#discussion_r628725653) is your comment from the pull request.

Reply:
I think we can leave this with `has('nvim-0.4')`. 0.5 is already released, and that's should be universal way to check version.

If we encounter that bug with git versions again, we could fix it again (or report to neovim).